digital-ocean: revert "make it easier to use disko"

schra opened the pull request:
This reverts commit d87c4e1a720f827996c282d0e194850a8903cebe from @Mic92

After switching from nixos-23.11 to nixos-unstable, I got the following error:

```
$ nixos-rebuild switch --flake nixos/#digitalocean
...
error:
Failed assertions:
- You must set the option ‘boot.loader.grub.devices’ or 'boot.loader.grub.mirroredBoots' to make the system bootable.
```

In my nixos config I don't set boot.loader.grub.devices or boot.loader.grub.mirroredBoots explicitly. So it seems to me like the values are not really defaulted via mkDefault? I am still relatively new to Nix, not sure what exactly is going on and how to fix this properly. I propose to revert this change.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Mic92 commented on nixos/modules/virtualisation/digital-ocean-config.nix:
```diff
@@ -41,7 +41,7 @@ with lib;
         initrd.kernelModules = [ "virtio_scsi" ];
         kernelModules = [ "virtio_pci" "virtio_net" ];
-        loader.grub.devices = lib.mkDefault ["/dev/vda"];
+        loader.grub.devices = ["/dev/vda"];
```
To fix the error that you see, we only need this:

```suggestion
        loader.grub.devices = ["/dev/vda"];
```

Mic92 replied:
The issue is this line: https://github.com/NixOS/nixpkgs/blob/e73004a744e8e8244d2e31c10931aad9610ec5d7/nixos/modules/system/boot/loader/grub/grub.nix#L716

schra replied:
Okay updated the PR to only update the one affected line.

> The issue is this line overriding our current default: 

Do you think it makes sense to reverse the mkDefault change instead of cleanly fixing it in grub.nix?

Mic92 replied:
Longterm probably but let's fix the regression first. The other fix also requires deprecation of the device option.

schra committed:
digital-ocean: revert "make it easier to use disko"

This reverts parts of commit d87c4e1a720f827996c282d0e194850a8903cebe from @Mic92

After switching from nixos-23.11 to nixos-unstable, I got the following error:

```
$ nixos-rebuild switch --flake nixos/#digitalocean
...
error:
Failed assertions:
- You must set the option ‘boot.loader.grub.devices’ or 'boot.loader.grub.mirroredBoots' to make the system bootable.
```

In my nixos config I don't set boot.loader.grub.devices or boot.loader.grub.mirroredBoots explicitly.

It seems like the values are not really defaulted via mkDefault? I am still relatively new to Nix, not sure how to fix this properly, so I am proposing to revert this change.
